pyomo/contrib/doe/doe.py

```diff
@@ -811,51 +589,44 @@ def _finite_calculation(self, output_record, scena_gen):
 
     def _extract_jac(self, m):
         """
-        Extract jacobian from simultaneous mode
+        Extract jacobian from the stochastic program 
         Arguments
         ---------
-        m: solved simultaneous model
+        m: solved stochastic program model
         Returns
         ------
         JAC: the overall jacobian as a dictionary
         """
         # dictionary form of jacobian
         jac = {}
         # loop over parameters
-        for p in list(self.param.keys()): 
+        for p in self.param.keys(): 
             jac_para = []
-            for name1 in self.jac_involved_name:
-                for tim in self.timepoint_overall_set:
-                    jac_para.append(pyo.value(m.jac[name1, p, tim]))
+            for res in m.res:
+                jac_para.append(pyo.value(m.jac[p, res]))
             jac[p] = jac_para
         return jac
 
-    def run_grid_search(self, design_values, design_ranges, design_dimension_names, 
-                    design_control_time, mode='sequential_finite', tee_option=False, 
+    def run_grid_search(self, design_ranges, design_dimension_names, 
+                     mode='sequential_finite', tee_option=False, 
                     scale_nominal_param_value=False, scale_constant_value=1, store_name= None, read_name=None,
                         filename=None, formula='central', step=0.001):
         """
         Enumerate through full grid search for any number of design variables;
         solve square problems sequentially to compute FIMs.
-        It calculates FIM with sensitivity information from four modes:
+        It calculates FIM with sensitivity information from two modes:
             1.  sequential_finite: Calculates a one scenario model multiple times for multiple scenarios. 
             Sensitivity info estimated by finite difference
-            2.  sequential_sipopt: calculate sensitivity by sIPOPT [Experimental]
-            3.  sequential_kaug: calculate sensitivity by k_aug [Experimental]
-            4.  direct_kaug: calculate sensitivity by k_aug with direct sensitivity
+            2.  direct_kaug: calculate sensitivity by k_aug with direct sensitivity
 
         Parameters
         -----------
-        design_values:
-            a ``dict`` where keys are design variable names, values are a dict whose keys are time point and values are the design variable value at that time point
         design_ranges:
             a ``list`` of design variable values to go over
         design_dimension_names:
             a ``list`` of design variable names of each design range
-        design_control_time:
-            a ``list`` of control time points that should be fixed to the values in dv_ranges
         mode:
-            use mode='sequential_finite', 'sequential_sipopt', 'sequential_kaug', 'direct_kaug'
+            use mode='sequential_finite', 'direct_kaug'
```

Comment:
Update this doc string to use Enum. See comments above for suggestions/formatting.